ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1760,6 +1765,81 @@ IopGetParentIdPrefix(PDEVICE_NODE DeviceNode,
     return Status;
 }
 
+static
+ULONG
+IopValidateID(IN PWCHAR Id,
+              IN SIZE_T MaxIdLen,
+              IN ULONG MaxSeparators,
+              IN BOOLEAN IsMultiSz)
+{
+   PWCHAR PtrSymbol;
+   ULONG_PTR StringEnd;
+   WCHAR Symbol;
+   ULONG SeparatorsCount = 0;
+   PWCHAR PtrPrevSymbol = NULL;
+
+   PAGED_CODE();
+
+   StringEnd = (ULONG_PTR)Id + MaxIdLen * sizeof(WCHAR);
```

Comment:
`StringEnd = Id + MaxIdLen;` , since MaxIdLen is a number of characters (I think you should just rename it to "cchIdLen" or just "IdLen", that's up to you).

Reply:
ok

Reply:
cch - ?